Library/Source/NativeEngine.cpp

```diff
@@ -1188,14 +1254,44 @@ namespace babylon
 
     Napi::Value NativeEngine::GetRenderWidth(const Napi::CallbackInfo& info)
     {
-        // TODO CHECK: Is this not just the size?  What is this?
-        return Napi::Value::From(info.Env(), m_renderTargetSize.Width);
+        return Napi::Value::From(info.Env(), bgfx::getStats()->width);
     }
 
     Napi::Value NativeEngine::GetRenderHeight(const Napi::CallbackInfo& info)
     {
-        // TODO CHECK: Is this not just the size?  What is this?
-        return Napi::Value::From(info.Env(), m_renderTargetSize.Height);
+        return Napi::Value::From(info.Env(), bgfx::getStats()->height);
+    }
+
+    void NativeEngine::SetViewPort(const Napi::CallbackInfo& info)
+    {
+        const auto x = info[0].As<Napi::Number>().FloatValue();
+        const auto y = info[1].As<Napi::Number>().FloatValue();
+        const auto width = info[2].As<Napi::Number>().FloatValue();
+        const auto height = info[3].As<Napi::Number>().FloatValue();
+
+        const auto backbufferWidth = bgfx::getStats()->width;
+        const auto backbufferHeight = bgfx::getStats()->height;
+        auto newViewId = m_viewidSet.Get();
+        m_viewportIds.push_back(newViewId);
+        m_currentBackbufferViewId = newViewId;
+        bgfx::setViewFrameBuffer(m_currentBackbufferViewId, BGFX_INVALID_HANDLE);
+        bgfx::setViewClear(m_currentBackbufferViewId, BGFX_CLEAR_COLOR | BGFX_CLEAR_DEPTH, 0x443355FF, 1.0f, 0);
+        if (bgfx::getCaps()->originBottomLeft)
```

Comment:
seem like most of this code is the same except for the y flip? maybe only branch for that one line?